.github/workflows/release.yml

```diff
@@ -0,0 +1,114 @@
+# Create a new GitHub release on pushed tag stating with 'v'.
+# Run tests, build and uploads binaries.
+
+name: Release
+
+on:
+  push:
+    tags:
+    - 'v*'
+
+jobs:
+
+  test:
```

Comment:
I'm new to Github Actions: Why do we need to test in both workflows?

Reply:
There are two workflows here. The `test` workflow is triggered on merging PRs to master. The `release` workflow is triggered on pushing a release tag. I have added the tests to the release as well because you could otherwise create a broken release.

If maybe you see a better solution to this we can switch.

Reply:
You could also run the workflow on push to tags and PRs for master. Then for individual jobs you can have a condition like

`if: startsWith( github.event.ref, 'refs/tags/v' )`

Reply:
Conceptually, I would like to separate test and release. Putting both in a dedicated Workflow seems okay for me. The problem is that our way of releasing (by pushing a tag) allows pushing untested code. Maybe there is a way to get rid of this, or we just say that is another problem and don't test.

Reply:
@awendt Would the suggestion by @fmasuhr seem okay for you? If so, I would try to switch to it and use one workflow with more conditionals.

Reply:
sounds good to me.